src/stan_math_backend/Transform_Mir.ml

```diff
@@ -395,6 +395,65 @@ let%expect_test "insert before" =
   [%sexp (l : int list)] |> print_s ;
   [%expect {| (1 2 3 4 5 999 6) |}]
 
+let rec add_validate_dims outvars stmts =
+  let params =
+    List.filter_map
+      ~f:(function
```

Comment:
Could you give this function a name? or the resulting `params` thing something indicating which params it should contain.

Reply:
It should contain all params whose constraint requires extra validation (simplex, unit vector, cholesky factor). But filtering the other params out was just a premature optimization so I changed it to just include all params.